src/unix/linux_like/linux/mod.rs

```diff
@@ -597,6 +624,97 @@ cfg_if! {
             }
         }
 
+        #[cfg(libc_union)]
+        #[allow(unused_unsafe)]
+        impl PartialEq for sockaddr_pppox {
+            fn eq(&self, other: &Self) -> bool {
+                self.sa_family == other.sa_family &&
+                self.sa_protocol == other.sa_protocol &&
+                match self.sa_protocol {
+                    /* PX_PROTO_OE */
+                    0 => unsafe { self.sa_addr.pppoe == other.sa_addr.pppoe },
+
+                    /* PX_PROTO_PPTP */
+                    2  => unsafe { self.sa_addr.pptp == other.sa_addr.pptp },
+
+                    _ => {
+                        debug_assert!(
+                            false,
+                            "The sockaddr_pppox only supports PX_PROTO_OE \
+                            and PX_PROTO_PPTP. sa_protocol was: {}",
+                            unsafe { self.sa_protocol.clone() }
+                        );
+                        false
+                    }
+                }
+            }
+        }
+        #[cfg(libc_union)]
+        impl Eq for sockaddr_pppox {}
+
+        #[cfg(libc_union)]
+        #[allow(unused_unsafe)]
+        impl ::hash::Hash for sockaddr_pppox {
+            fn hash<H: ::hash::Hasher>(&self, s: &mut H) {
+                unsafe {
+                    self.sa_family.hash(s);
+                    self.sa_protocol.hash(s);
+                }
+
+                match self.sa_protocol {
+                    /* PX_PROTO_OE */
+                    0 => unsafe { self.sa_addr.pppoe.hash(s) },
+
+                    /* PX_PROTO_PPTP */
+                    2 => unsafe { self.sa_addr.pptp.hash(s) },
+
+                    _ => {
+                        debug_assert!(
+                            false,
+                            "The sockaddr_pppox only supports PX_PROTO_OE \
+                            and PX_PROTO_PPTP. sa_protocol was: {}",
+                            unsafe { self.sa_protocol.clone() }
+                        );
+                    }
+                }
+            }
+        }
+
+        #[cfg(libc_union)]
+        impl ::fmt::Debug for sockaddr_pppox_sa_addr_t {
+            fn fmt(&self, f: &mut ::fmt::Formatter) -> ::fmt::Result {
+                assert_eq!(24, ::mem::size_of::<Self>());
+                unsafe { ::mem::transmute::<_, [u8; 24]> (*self).fmt(f) }
+            }
+        }
```

Comment:
Hm, I think we should the same way as well as other places.

Reply:
fixed